### ISSUE TITLE
Before and After hooks are placed wrong in the test runner

### ISSUE THREAD

**Issue by Konstruktour** [cypress-io/cypress#2663]
### Current behavior:

The following test structure ...
```
describe('describe 1', () => {
    before('before all 1', () => {
        cy.log('before all 1');
    });

    beforeEach('before each 1', () => {
        cy.log('before each 1');
    });

    it('it A', () => {
        cy.log('it A')
    });

    it('it B', () => {
        cy.log('it B')
    });

    afterEach('after each 1', () => {
        cy.log('after each 1');
    });

    after('after all 1', () => {
        cy.log('after all 1');
    });
});

```

... results in the following view in the cypress test runner gui

![image](https://user-images.githubusercontent.com/7323845/47555316-c5ba7900-d90b-11e8-8101-a7c7486996cd.png)

The `before` hook is rendered into the first `it` hook. The `after` hook into the last `it` hook.  

### Desired behavior:
The `before` and `after` hooks should not be part of the `it` hook, they should placed inbetween describe and it.
The description of the hooks are missing in the cypress test runner gui.

### Versions
Cypress 3.1.0
Windows 10
Electron 59

**Comment by jennifer-shehane** [comment]
Yeah, I can see what you mean. The `before` and `after` hook are accurate in terms of the placement and order in which they are run, but could definitely cause confusion in not knowing where exactly to look for `before` or `after` hooks displayed. 

I will say this would be quite a low priority in our roadmap unfortunately, but a description of the problem this caused you would help if you can provide it.

**Comment by Konstruktour** [comment]
Another issue, which is sometimes a little bit annoying - all before / after hooks are (visually) merged together. 
eg
```
describe('describe 1', () => {
    before('before all 1', () => {
        cy.log('before all 1');
    });

    before('before all 2', () => {
        cy.log('before all 2');
    });

    it('it A', () => {
        cy.log('it A')
    });

    it('it B', () => {
        cy.log('it B')
    });

    after('after all 1', () => {
        cy.log('after all 1');
    });

    after('after all 2', () => {
        cy.log('after all 2');
    });
});
```
results in 

![image](https://user-images.githubusercontent.com/7323845/47586148-a8af9580-d95f-11e8-844c-40cd68adb1c0.png)

it would be easier for debugging if the hooks are displayed "correctly" (also including the description). think it is better if they follow the same structure like in the code.

**Comment by jennifer-shehane** [comment]
Related issue - if first test is skipped, no global hooks are logged https://github.com/cypress-io/cypress/issues/1846
